be/src/olap/memtable.cpp

```diff
@@ -302,7 +302,7 @@ Status MemTable::_do_flush(int64_t& duration_ns) {
         }
     } else {
         vectorized::Block block = _collect_vskiplist_results();
-        RETURN_NOT_OK(_rowset_writer->add_block(&block));
+        RETURN_NOT_OK(_rowset_writer->flush_single_memtable(&block));
         _flush_size = block.allocated_bytes();
         RETURN_NOT_OK(_rowset_writer->flush());
```

Comment:
flush method is flushing _segment_writer, not the writer created in flush_single_memtable, maybe we could call flush(segment writer) at flush_single_memtable method.